File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character,Integer> mp=new HashMap<>();
        int left=0;
        int max=0;
        for(int r=0;r<s.length();r++){
            char c=s.charAt(r);
            if(mp.containsKey(c)){
                left=Math.max(left,mp.get(c)+1);
            }
            mp.put(c,r);
            max=Math.max(max,r-left+1);
        }
        return max;
        
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<bool> vis(128,false);
        int n=s.size();
        int mx=0;
        int l=0;
        for(int r=0;r<n;r++){
            while(vis[s[r]]){
                vis[s[l]]=false;
                l++;
            }
            vis[s[r]]=true;
            mx=max(mx,r-l+1);
        }
        return mx;
    }
};